### PR TITLE
Turn off rx for all interfaces to improve performance

### PR DESCRIPTION
**Pull request by cs-zhang**
Signed-off-by: CS Zhang <zhangcho@us.ibm.com>

### CODE REVIEW

**Comment by rpsene** [comment]
` Error: Reference to undeclared input variable
│
│   on modules/1_prepare/prepare.tf line 164, in resource "null_resource" "bastion_init":
│  164: if [[ ${var.fips_compliant} = true ]]; then
│
│ An input variable with the name "fips_compliant" has not been declared. This variable can be declared with a variable "fips_compliant" {}
│ block.`

@cs-zhang got this error ^

**Comment by cs-zhang** [comment]
The `var.fips_compliant` not defined error seems like your did not patch the right release branch. My update did not add any new variables.

**Comment by rpsene** [comment]
> The `var.fips_compliant` not defined error seems like your did not patch the right release branch. My update did not add any new variables.

I applied it no 4.10 branch, where FIPS variables are available.

**Comment by cs-zhang** [comment]
FIPS for bastion just on master only

**Comment by cs-zhang** [comment]
@yussufsh @Prajyot-Parab can you review it? @rpsene wants it ASP.

**Review by sajauddin — COMMENTED** [review]
/lgtm

File: modules/1_prepare/prepare.tf [hunk 1]
```diff
@@ -140,25 +140,35 @@ resource "null_resource" "bastion_init" {
...
+sudo sed -i.bak -e 's/^ - set_hostname/# - set_hostname/' -e 's/^ - update_hostname/# - update_hostname/' /etc/cloud/cloud.cfg
+sudo hostnamectl set-hostname --static ${lower(var.name_prefix)}bastion-${count.index}.${var.cluster_domain}
+echo 'HOSTNAME=${lower(var.name_prefix)}bastion-${count.index}.${var.cluster_domain}' | sudo tee -a /etc/sysconfig/network > /dev/null
+sudo hostname -F /etc/hostname
+echo 'vm.max_map_count = 262144' | sudo tee --append /etc/sysctl.conf > /dev/null
+
+# Set SMT to user specified value; Should not fail for invalid values.
+sudo ppc64_cpu --smt=${var.rhel_smt} | true
+
+# turn off rx and set mtu to 1450 for all ineterfaces to improve network performance
```
[the remark this answers is not in the document]
Comment (rpsene) [reply modules/1_prepare/prepare.tf:155]: @yussufsh we need on both interfaces; public and private.

File: modules/1_prepare/prepare.tf [hunk 2]
```diff
@@ -140,25 +140,35 @@ resource "null_resource" "bastion_init" {
...
+cidrs=("${ibm_pi_network.public_network.pi_cidr}" "${data.ibm_pi_network.network.cidr}")
+for cidr in "$${cidrs[@]}"; do
+  envs=($(ip r | grep "$cidr dev" | awk '{print $3}'))
+  for env in "$${envs[@]}"; do
+    con_name=$(sudo nmcli -t -f NAME connection show | grep $env)
+    sudo nmcli connection modify "$con_name" ethtool.feature-rx off
+    sudo nmcli connection modify "$con_name" ethernet.mtu 1450
```
[the remark this answers is not in the document]
Comment (cs-zhang) [reply modules/1_prepare/prepare.tf:162]: can we use `private_network_mtu` for public interface too?

File: modules/1_prepare/prepare.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (cs-zhang) [reply modules/1_prepare/prepare.tf]: these code no longer required, it has been removed.

File: modules/1_prepare/prepare.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (yussufsh) [reply modules/1_prepare/prepare.tf]: I would name it as simply `network_mtu`.

[the remark this answers is not in the document]
Comment (yussufsh) [reply modules/1_prepare/prepare.tf]: > I would name it as simply `network_mtu`.

@cs-zhang Maybe you can do this later if this is an urgent change needed. Kindly update your branch and squash the commits and I will give it a 👍 .

**Comment by cs-zhang** [comment]
Squashed the commits to one for now, late we can use `network_mtu`  to replace `private_network_mtu`.

**Review by yussufsh — APPROVED** [review]
/lgtm

**Comment by yussufsh** [comment]
@cs-zhang please rebase this one.

**Comment by yussufsh** [comment]
@cs-zhang please rebase this one.

**Comment by cs-zhang** [comment]
@yussufsh @sajauddin rebased

**Review by yussufsh — APPROVED** [review]

**Review by sajauddin — COMMENTED** [review]
/lgtm

**Review by sajauddin — APPROVED** [review]

**Pull request merged**